src/api/Shrooms.Domain/Services/Wall/WallService.cs

```diff
@@ -767,6 +768,7 @@ private async Task<IList<WallDto>> GetAllOrNotFollowedWallsAsync(UserAndOrganiza
             var wallFilters = new Dictionary<WallsListFilter, Expression<Func<MultiwallWall, bool>>>
             {
                 { WallsListFilter.All, w => true },
+                { WallsListFilter.NotHidden, w => !w.IsHidden || (w.IsHidden && w.Members.Any(m => m.UserId == userOrg.UserId)) },
```

Comment:
`w.IsHidden` is redundant it should be enough to check `!w.IsHidden || w.Members.Any(m => m.UserId == userOrg.UserId)`